src/Functions/UTCTimestampTransform.cpp

```diff
@@ -0,0 +1,184 @@
+                    date_time_col->get(i, date_time_val);
+                    time_zone_col->get(i, time_zone_val);
+                    std::string time_zone = time_zone_val.get<String>();
+                    LocalDateTime date_time(date_time_val.get<UInt32>(), Name::to ? DateLUT::instance("UTC") : DateLUT::instance(time_zone));
+                    time_t utc_time_val = date_time.to_time_t(Name::from ? DateLUT::instance("UTC") : DateLUT::instance(time_zone));
+                    column->insert(Field(utc_time_val));
+                }
+            }
+            else if (WhichDataType(arg1.type).isDateTime64())
+            {
+                const auto * date_time_col = checkAndGetColumn<ColumnDateTime64>(arg1.column.get());
+                if (!date_time_col)
+                {
+                    const auto * date_time_const_col = checkAndGetColumn<ColumnConst>(arg1.column.get());
+                    if (!date_time_const_col)
+                        throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 1st argument is not datatime64 type or const type.", name);
+                    date_time_col = checkAndGetColumn<ColumnDateTime64>(date_time_const_col->getDataColumn());
+                    if (!date_time_col)
+                        throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 1st argument must be datetime64 type", name);
+                }
+                const DataTypeDateTime64 * date_time_type = static_cast<const DataTypeDateTime64 *>(arg1.type.get());
+                Int64 scale_multiplier = DecimalUtils::scaleMultiplier<Int64>(date_time_type->getScale());
+                for (size_t i = 0; i < input_rows_count; ++i)
+                {
+                    Field date_time_val;
+                    Field time_zone_val;
+                    date_time_col->get(i, date_time_val);
+                    time_zone_col->get(i, time_zone_val);
+                    std::string time_zone = time_zone_val.get<String>();
+                    DateTime64 date_time_64 = date_time_val.get<DateTime64>();
+                    Int64 seconds = date_time_64.value / scale_multiplier;
+                    Int64 micros = date_time_64.value % scale_multiplier;
+                    LocalDateTime date_time(seconds, Name::to ? DateLUT::instance("UTC") : DateLUT::instance(time_zone));
+                    time_t utc_time_val = date_time.to_time_t(Name::from ? DateLUT::instance("UTC") : DateLUT::instance(time_zone));
+                    DateTime64 utc_time(utc_time_val * scale_multiplier + micros);
+                    column->insert(Field(utc_time));
+                }
+            }
+            else
+                throw Exception(ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT, "Function {}'s 1st argument can only be datetime/datatime64. ", name);
+            return column;
+        }
+
+    };
+
+    struct NameToUTCTimestamp
+    {
+        static constexpr auto name = "to_utc_timestamp";
```

Comment:
It will be better to use `camelCase` for function names. If you still want to have names with underlines, add them as aliases